test/test_type_hints.py

```diff
@@ -93,30 +100,28 @@ def get_all_examples():
 
 
 class TestTypeHints(TestCase):
-    @unittest.skipIf(sys.version_info[0] == 2, "not type hints for Python 2")
+    @unittest.skipIf(sys.version_info[0] == 2, "no type hints for Python 2")
     @unittest.skipIf(not HAVE_MYPY, "need mypy")
     def test_doc_examples(self):
-        """run documentation examples through mypy.
-mypy can be picky about its environment, so we need
-to set up a temporary dir and link the torch module
-into it"""
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            fn = os.path.join(tmp_dir, 'test.py')
-            curdir = os.getcwd()
-            os.chdir(tmp_dir)
-            with open(fn, "w") as f:
-                print(get_all_examples(), file=f)
-            try:
-                os.symlink(os.path.dirname(torch.__file__), './torch', target_is_directory=True)
-            except OSError:
-                raise unittest.SkipTest('cannot symlink')
-            try:
-                result = subprocess.run(['python3', '-mmypy', '--follow-imports',
-                                         'silent', '--check-untyped-defs', 'test.py'],
-                                        check=True)
-            except subprocess.CalledProcessError as e:
-                raise AssertionError("mypy failed")
-            os.chdir(curdir)
+        """
+        Run documentation examples through mypy.
+        """
+        fn = os.path.join('test', 'data', 'type_hints_smoketest.py')
+        with open(fn, "w") as f:
+            print(get_all_examples(), file=f)
+        try:
+            # TODO: I'm not sure we actually need to start a new process
```

Comment:
What I ran into trouble with was to make sure mypy picks up the right PyTorch. (Which is why I did this temporary directory dance.)

Reply:
My usual workflow is either
```
python3 setup.py bdist_wheel
pip3 install dist/*.whl
pyton3 test/test_abc.py
```
or
```
python3 setup.py build
PYTHONPATH=./build/lib.<something/ pyton3 test/test_abc.py
```
one of them didn't work because mypy would either not complain enough or too much. (So I do want it to fail if there is funny stuff in the examples.)

Now that *could* be for totally unrelated reasons (bug in the mypy version I used back then, command line options that I only added later, ...), so if you find that it works for you, I'm more than happy to not use kludges like the temporary directory that.

So I'd say drop it now and if I stumble over it again, we can see again.